backend/app/dependencies.py

```
@@ -0,0 +1,135 @@
+    """Require admin role for access"""
+    if current_user.role != "admin":
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Not enough permissions"
+        )
+    return current_user
+
+
+async def get_current_user_optional(
+    request: Request,
+    db: Session = Depends(get_db)
```

Ruff (0.14.5)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)